src/treeutils/optimise.jl

```diff
@@ -0,0 +1,249 @@
+"""
+circuit optimisation
+"""
+module Optimise
```

Comment:
I think we will move this module to a separate package instead of developing it inside YaoBlocks eventually (but reexport it in Yao), so YaoBlocks can be more stable. Maybe we just do this in next release along with  the remove of `MathGate` etc?

Reply:
I'd prefer a submodule `YaoBlocks.Optimise` like `YaoBlocks.AD`. I have been suffering a lot from seperating to many packages in order to keep them consistent. Agree?

Reply:
`Optimise` and `AD` are important, and closely related to `YaoBlocks`. Unlike `YaoExtensions` that providing utilities, they can break on any change of `YaoBlocks`.